op/model.py

```diff
@@ -302,3 +335,10 @@ def relation_set(self, relation_id, key, value):
 
     def config_get(self):
         return self._run('config-get')
+
+    def resource_get(self, resource_name):
+        try:
+            return self._run_text('resource-get', resource_name).strip()
+        except CalledProcessError as e:
+            # The resource was not attached (local charm) or could not be fetched from the controller.
+            raise ResourceError(e.stderr.decode('utf8'))
```

Comment:
We had a good conversation about handling of errors on Friday. We should repeat it in the standup tomorrow so we're all in the same wave length. The very terse summary is that there's little value in creating such error upfront and simply bubbling up the message from the underlying _run method. That will lead to several error types, which perhaps won't capture the errors we do want to handle. As an alternative, we can let _run itself raise the error generically, and then implement error types for the things we do find out require custom handling.

Reply:
From my experience, handling this case will be a necessity for charms.  I was actually on the fence about having this just set the value to `None` instead of raising the error, but I think making the fetch explicit and having that return a success flag is even better.

Reply:
My experience also shows that handling errors is a necessity.. the question is which errors do we need to handle, and how. For example, we might try to handle all of them, but that's pretty much impossible because we don't know which errors the other systems we talk to (including the juju agent) will raise in the future. We can also try to handle them in groups, but we don't quite have enough information just yet about which groups are these. As such, my suggestion is for us to have a consistent and general way to handle errors right now, and then become more fine grained as we learn about the points in which people want to distinguish between independent kind of errors.